compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1330,7 +1330,8 @@ private void checkErrorsOfUserDefinedType(SymbolEnv env, BLangNode unresolvedTyp
             if (typeDefinitions.isEmpty()) {
                 BType referredType = symResolver.resolveTypeNode(currentTypeOrClassNode, env);
                 if (referredType.tag == TypeTags.RECORD || referredType.tag == TypeTags.OBJECT ||
-                        referredType.tag == TypeTags.FINITE) {
+                        referredType.tag == TypeTags.FINITE || (referredType.tag == TypeTags.UNION &&
```

Comment:
It is enough to add `referredType.tag == TypeTags.UNION` here. But this whole condition is not optimum. Ideally 
`referredType != symTable.semanticError && referredType != symTable.noType` kind of logic should work right.

Reply:
Check this also https://github.com/ballerina-platform/ballerina-lang/issues/30101#issuecomment-863888227